pkg/hubble/parser/seven/parser.go

```diff
@@ -176,10 +177,9 @@ func (p *Parser) computeResponseTime(r *accesslog.LogRecord, timestamp time.Time
 func (p *Parser) updateEndpointWorkloads(ip net.IP, endpoint *pb.Endpoint) {
 	if ep, ok := p.endpointGetter.GetEndpointInfo(ip); ok {
 		if pod := ep.GetPod(); pod != nil {
-			olen := len(pod.GetOwnerReferences())
-			endpoint.Workloads = make([]*pb.Workload, olen)
-			for index, owner := range pod.GetOwnerReferences() {
-				endpoint.Workloads[index] = &pb.Workload{Kind: owner.Kind, Name: owner.Name}
+			workload, workloadTypeMeta, ok := utils.GetWorkloadMetaFromPod(pod)
```

Comment:
What happens if the pod belongs to a StatefulSet?

Reply:
I'll add a test for that. It should fallback to the controller ref, which for a pod managed by a StatefulSet is the StatefulSet.

Reply:
Added a new test in the latest changes. ✔️